src/domain/entities/UserLogic.ts

```diff
@@ -14,4 +14,10 @@ export class UserLogic extends Struct<User>() {
             user => user.organisationUnits.length > 0 && user.userRoles.length > 0 && user.userGroups.length > 0
         );
     }
+
+    static validateUniqueOpenId(users: User[], allUsers: User[]): boolean {
+        const openIds = users.map(user => user.openId);
+        const allOpenIds = allUsers.map(user => user.openId);
+        return openIds.every(openId => !allOpenIds.includes(openId));
```

Comment:
This works, but since this is a generic app we could have a server with thousands of users. Let's avoid having a nested loop (every -> includes). We can improve the validation by using a set:

```ts
static validateUniqueOpenId(users: User[], allUsers: User[]): boolean {
  const allOpenIdsSet = new Set(allUsers.map(user => user.openId));
  return users.every(user => !allOpenIdsSet.has(user.openId));
}
```